Fixes #1318 by multiplying decreaseExpirationBySec with 1000. Also fixes returning true for calling hasValidAccessToken on tokens which are already expired

IchbinkeinReh opened the pull request:
In the "expires_at" is set and its calculated with local clock (see https://github.com/manfredsteyer/angular-oauth2-oidc/blob/15.0.0/projects/lib/src/oauth-service.ts#L1698 ). Therefore there is no need to appy clockSkewInMsec in hasValidAccessToken

IchbinkeinReh committed:
Fixes #1318 by multiplying decreaseExpirationBySec with 1000. Also fixes wrong subtraction of getClockSkewInMsec

Signed-off-by: Patrick Bender <Patrick.Bender@softgarden.de>

jeroenheijmans commented:
Heya! FYI: I'm no longer very active as a maintainer or community moderator (see #1280), so I will likely not be investigating or reviewing. But thanks for submitting a fix to this Open Source project, hopefully Manfred or another contributor can find time to help out!

IchbinkeinReh commented:
@manfredsteyer sry for bumping, but i think this is a big issue and should be fixed asap

bkrajendra commented:
Need to be merged.

vilmantasbaranauskas commented:
The PR fixes 2 occurrences of decreaseExpirationBySec but there are 3 occurrences on current master.

:+1:   to merge it.

IchbinkeinReh commented:
> The PR fixes 2 occurrences of decreaseExpirationBySec but there are 3 occurrences on current master.
> 
> 👍 to merge it.

Thx. I just rebased to latest master and also fixed the third occurence

vilmantasbaranauskas reviewed (approved):
PR looks correct to me.
